app/services/electric_vehicle.py

```diff
@@ -135,3 +136,34 @@ def get_vehicle_list(
         "list_edge": edges,
     }
     return results
+
+
+def get_by_id(id, db, current_user):
+    query = db.query(Vehicle)
+    if not check_role_supervisor(current_user):
+        company_names = get_company_name_from_user(current_user, db)
+        query = query.join(SaleInformation, Customer, Company).filter(
+            Vehicle.sale_id == SaleInformation.id,
+            SaleInformation.customer_id == Customer.id,
+            Customer.company_id == Company.id,
+            Company.name.in_(company_names),
+            Vehicle.id == id,
+        )
+    else:
+        query = query.filter(
+            Vehicle.id == id,
+        )
+    return query
+
+
+def get_detail(id, db, current_user):
+    vehicle = get_by_id(id, db, current_user).first()
+    if vehicle:
```

Comment:
if not vehicle: 
    raise

return {
            "detail": jsonable_encoder(vehicle),
            "charger": vehicle.charger,
            "model": vehicle.vehicle_model,
        }